src/main/java/biz/paluch/logging/gelf/log4j2/Log4j2LogEvent.java

```diff
@@ -172,11 +173,19 @@ private String getValue(MdcMessageField field) {
     @Override
     public String getMdcValue(String mdcName) {
         ReadOnlyStringMap contextData = logEvent.getContextData();
-        if (null != contextData && contextData.containsKey(mdcName)) {
-            return contextData.getValue(mdcName);
+        if (null == contextData) {
```

Comment:
How about converting the value ourselves to `String`? Creating another map imposes quite significant allocations?